src/Type/Php/PregSplitDynamicReturnTypeExtension.php

```diff
@@ -61,6 +63,21 @@ private function hasFlag(int $flag, ?Arg $expression, Scope $scope): bool
 			return false;
 		}
 
+		if ($expression->value instanceof BitwiseOr) {
```

Comment:
IMO this `hasFlag` handling could be refactored into a helper since its usefull for extension development.

at least the builtin `FilterVarDynamicReturnTypeExtension` has a similar logic

Reply:
Just out of curiosity: adding support for more than 3 flags via bitwise or would involve recursive resolution of them I guess? Basically getting all the leaves from the bitwise expression tree.

Reply:
I am fine with whatever we agree on. I can work on it.

Reply:
I don't see it being recursive in this PR. Can you please modify it? Thanks.

Reply:
Good catch. Fixed and rebased